### PR TITLE
Added logic to strip any existing hyphens before processing the args.

### PR DESCRIPTION
**Pull request by phillipsj**
Updated the logic to handle if extra args are passed with existing hyphens in the arg. The test was updated to add the additional case of having pre-existing hyphens. The method name was also refactored based on previous feedback.

* https://github.com/k3s-io/k3s/issues/3642

### CODE REVIEW

**Review by briandowns — APPROVED** [review]

File: pkg/daemons/config/types.go [hunk 1]
```diff
@@ -258,10 +258,13 @@ func (a ArgString) String() string {
 	return b.String()
 }
 
-func GetArgsList(argsMap map[string]string, extraArgs []string) []string {
+// GetArgs appends extra arguments to existing arguments overriding any default options.
+func GetArgs(argsMap map[string]string, extraArgs []string) []string {
+	const hyphens = "--"
```
Comment (brandond) [inline pkg/daemons/config/types.go:263]: nit: we don't generally use consts inside functions in this codebase; move this up into the top of this file.

File: pkg/daemons/config/types.go [hunk 2]
```diff
@@ -270,7 +273,7 @@ func GetArgsList(argsMap map[string]string, extraArgs []string) []string {
 	}
 	var args []string
 	for arg, value := range argsMap {
-		cmd := fmt.Sprintf("--%s=%s", arg, value)
+		cmd := fmt.Sprintf("--%s=%s", strings.TrimPrefix(arg,hyphens), value)
```
Comment (brandond) [inline pkg/daemons/config/types.go:276]:
```suggestion
		cmd := fmt.Sprintf("%s%s=%s", hyphens, strings.TrimPrefix(arg,hyphens), value)
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: pkg/daemons/config/types.go (hunk 1 above)
Comment (briandowns) [reply pkg/daemons/config/types.go:263]: If it's only used in this function, it's convention to keep it within the scope it's used.

**Comment by brandond** [comment]
It looks like you merged master into your branch, instead of rebasing onto it. Can you switch that around?

**Review by brandond — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Pull request merged**